### PR TITLE
[@types/stripe] createSource accepts bank account arguments

### PR DESCRIPTION
**Pull request by danlite**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: (https://stripe.com/docs/api/customer_bank_accounts/create?lang=node)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/stripe/stripe-tests.ts [hunk 1]
```diff
@@ -635,6 +635,41 @@ stripe.customers.createSource(
...
+            currency: 'USD',
+            account_holder_name: 'Account Holder',
+            account_holder_type: 'individual',
+            account_number: '000123456789',
+            routing_number: '110000000'
+        }
+    }).then(
+    (bankAcc) => {
+        // asynchronously called
+        const obj: Stripe.IBankAccount = bankAcc as Stripe.IBankAccount;
```
Comment (vaskevich) [inline types/stripe/stripe-tests.ts]: Can you change this to `bankAcc; // $ExpectType IBankAccount`? The current casts won't do a ton of good.

**Review by vaskevich — APPROVED** [review]
LGTM

**Comment by danlite** [comment]
I made the recommended change, and then it turns out the method signature I intended (provide bank account source, return `IBankAccount`) wasn't being enforced, so good catch.

However, I can't get tests to pass now, with the error:
```
ERROR: 922:9  expect  TypeScript@next expected type to be:
  IList<ICard | IBankAccount>
got:
  IList<IBankAccount | ICard>
```

If I switch the order of the union, the (inverse) error occurs for TypeScript@3.0... I'll submit my broken commit in the hopes that someone can chime in.

**Comment by danlite** [comment]
For reference:
https://travis-ci.org/DefinitelyTyped/DefinitelyTyped/builds/575551389#L320

**Comment by vaskevich** [comment]
Uh oh. Looks like they started alphabetizing union types in `typescript@next`. Is there a reason that it's not returning just `IBankAccount`, though? Provided the parameters passed in are correct, TS should be able to infer the return type as just `IBankAccount` or `ICard`, depending on the parameters.

**Comment by danlite** [comment]
The weird thing is, it wasn't failing on the method I modified (`stripe.customers.createSource`), but rather on `stripe.accounts.retrieve` 🤔 

In any case, I've created a helper type that seems to resolve that issue. It feels a little hacky, so I'm open to other suggestions.

**Pull request merged**